realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -869,6 +870,50 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeFind(
     return -1;
 }
 
+JNIEXPORT jlong JNICALL Java_io_realm_internal_TableQuery_nativeFindWithHandover(
+    JNIEnv* env, jobject, jlong bgSharedGroupPtr, jlong handoverQueryPtr, jlong fromTableRow)
+{
+    try {
+      // cast the pointer
+      SharedGroup::Handover<Query>* hQuery = reinterpret_cast<SharedGroup::Handover<Query>* > (handoverQueryPtr);
+      std::unique_ptr<SharedGroup::Handover<Query> > handoverQuery (hQuery);
+
+      // when opening a Realm, we start implicitly a read transaction
+      // we need to close it before calling begin_read
+      SG(bgSharedGroupPtr)->end_read();
```

Comment:
No, this is a shared group created only for the purpose of the query we control it's entire lifecycle